src/ideas-behind-linuxboot.md

```diff
@@ -0,0 +1,87 @@
+megabytes or even gigabytes.
+
+We neglect other storages here that are attached via NVMe, SATA or other
+high-speed buses, because those are commonly not supported by mask ROMs. They
+are, on the other hand, what a bootloader offers to boot from, as well as
+network sources.
+
+### Embedded devices
+
+Many devices, nowadays known as IoT (Internet of Things), appliances, or
+similar, have a narrow use case. They are meant to perform a specific set of
+tasks, and thus can be tailored for it. In hardware terms, that often means an
+SoC, a bit of storage, and peripherals. Debug interfaces are reduced or removed
+for the final product.
+
+### Desktop, laptop, workstation and server systems
+
+At this point, many systems are still based on x86 processors, coming with a SPI
+flash on the board. While laptops and desktops mostly have a mere 16 or 32
+megabytes to offer, high-end servers and workstations already have 64, and even
+a second whole system called the Board Management Controller (BMC), which has
+its own firmware and corresponding storage. Designs around those constraints
+vary among OEMs.
+
+Note that it need not be that way. Arm based, RISC-V based and other systems
+already show that you can expect more, such as booting off eMMC. Laptops and
+desktop boards in that range are available as of now, even some servers and
+workstations.
+
+### Single Board Computers (SBCs)
+
+The SBC market has grown to such a degree that credit card size boards nowadays
+comes with both small storage parts that can act as boot sources as well as PCIe
+connectors that can hold NVMes of gigabytes and terabytes of storage. This gives
+us the opportunity to work out a boot flow that provides the end user with a
+very rich environment already early on before the main operating system is
+loaded.
+
+## Evaluation of tools
+
+There are many existing tools already that we can leverage to implement the idea
+of using Linux to boot into an operating system.
+
+### Root filesystem
+
+| tool | language | license | usage |
+| - | - | - | - |
+| [BusyBox](https://busybox.net/) | C | GPLv2 | Heads |
+| [toybox](http://landley.net/toybox) | C | 0BSD | Android |
+| [GNU coreutils](https://www.gnu.org/software/coreutils/) | C | GPLv3 | not for LinuxBoot |
+| [u-root](https://u-root.org) | Go | BSD 3-Clause | ByteDance, Google et al |
+| [uutils/coreutils](http://uutils.github.io/) | Rust | MIT | not for LinuxBoot |
+
+t.b.d
+
+### kexec implementations
+
+| tool | language | license | usage |
+| - | - | - | - |
+| [kexec-tools](https://git.kernel.org/pub/scm/utils/kernel/kexec/kexec-tools.git) ([GitHub mirror](https://github.com/horms/kexec-tools)) | C | GPLv2 | Heads, Petitboot |
+| [systemd](https://systemd.io/) (wrapper) | C | LGPL-2.1+ | systemd on UEFI |
+| [kexecboot](https://github.com/kexecboot/kexecboot) | C | GPLv2 | ? |
+| u-root (CLI+mod) | Go | BSD 3-Clause | Google et al |
+| [kexlinux](https://github.com/im-0/kexlinux) | LGPL-3.0+ | Rust | ? |
+
+t.b.d.
```

Comment:
I'd remove the `t.b.d.` since that could really be added throughout the whole book